### PR TITLE
refactor(SelectPicker): move grouped options into group element

### PR DESCRIPTION
**Pull request by SevenOutman**
Before: grouped option elements are siblings of group element

image

a11y inspector:
image

After: grouped option elements are within group element

image

a11y inspector:
image

### CODE REVIEW

**Review by simonguo — COMMENTED** [review]
The menus for SelectPicker and InputPicker are almost identical. Is it possible to make the components reusable?

File: src/SelectPicker/Listbox.tsx [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+          </List>
+        )}
+      </AutoSizer>
+    );
+  }, [listProps, maxHeight, options, renderOption, rowHeight, virtualizedListRef]);
+
+  // Example of rendering option groups in VariableSizeList
+  // https://github.com/bvaughn/react-window/issues/358
+  const renderVirtualizedOptionGroups = useCallback(() => {
+    const groups = groupOptions(options, groupBy!, sort?.(false), sort?.(true));
```
[the remark this answers is not in the document]
Comment (SevenOutman) [reply src/SelectPicker/Listbox.tsx]: Updated `renderVirtualizedOptionGroups` to take a `groupKey` argument in order to ensure it's defined.

File: src/SelectPicker/Listbox.tsx [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+        >
+          {renderOptions(group.options)}
+        </ListboxOptionGroup>
+      );
+    },
+    [foldedGroupKeys, handleGroupTitleClick, renderMenuGroup, renderOptions]
+  );
+
+  const renderOptionGroups = useCallback(() => {
+    const groups = groupOptions(options, groupBy!, sort?.(false), sort?.(true));
```
[the remark this answers is not in the document]
Comment (SevenOutman) [reply src/SelectPicker/Listbox.tsx]: Updated `renderOptionGroups ` to take a `groupKey` argument in order to ensure it's defined.

**Comment by SevenOutman** [comment]
> The menus for SelectPicker and InputPicker are almost identical. Is it possible to make the components reusable?

Yes, the Listbox component will be reused across picker components that have similar behavior. This edit only involves SelectPicker update in order to keep changes small.

**Review by simonguo — APPROVED** [review]

**Pull request merged**